### PR TITLE
fix: falsly values

### PR DESCRIPTION
**Pull request by pdrbrnd**
Doing this currently assumes that there is no value
```tsx
t('somekey', { value: 0 })
```

I think we should accept falsly values

### CODE REVIEW

**Comment by plckr** [comment]
the only benefit to allow falsly values is for an integer 0, right?
what would happen with a null value in this case ?

Also, isn't it better to just allow every type, even the undefined ? if it's undefined it'll resolve empty.
I could be wrong, but forcing the user to pass a non undefined variable seems strange for me

**Comment by TGlide** [comment]
> the only benefit to allow falsly values is for an integer 0, right? what would happen with a null value in this case ?
> 
> Also, isn't it better to just allow every type, even the undefined ? if it's undefined it'll resolve empty. I could be wrong, but forcing the user to pass a non undefined variable seems strange for me

You could also accept an empty string, although it wouldn't be super useful. About null, I think we should check for it too, meaning we accept everything except null or undefined. 

That brings me the question though: How do we deal with optional values? Say I want one translation key to have required values, and another to have optional ones? I don't think it's a common problem, but something to think about.

**Comment by TGlide** [comment]
Now, one thing to note, I'm curious about https://github.com/ivanhofer/typesafe-i18n. It's framework agnostic, and seems pretty robust, even allowing stuff such as custom value formatters.

**Comment by plckr** [comment]
> That brings me the question though: How do we deal with optional values? Say I want one translation key to have required values, and another to have optional ones? I don't think it's a common problem, but something to think about.

I like it, also syntax could be quite easy
`{{term?}}` - optional
`{{term}}` - required
or 
`{{term}}` - optional
`{{term!}}` - required

**Comment by TGlide** [comment]
> I like it, also syntax could be quite easy `{{term?}}` - optional `{{term}}` - required or `{{term}}` - optional `{{term!}}` - required

That would be aweeeesome

File: src/utils.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const parseMatchesForKey = <
 ): T => {
   for (const match of matches) {
     const matchName = match.replace(/\{\{|\}\}/gim, '')
-    if (!values[matchName]) {
+    if (values[matchName] === undefined) {
```
Comment (TGlide) [inline src/utils.ts:47]:
```suggestion
    if ([undefined, null].includes(values[matchName])) {
```

**Pull request merged**